Tests/NIOExtrasTests/LengthFieldPrependerTest.swift

```diff
@@ -0,0 +1,424 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import NIO
+import NIOExtras
+
+private let standardDataString = "abcde"
+
+class LengthFieldPrependerTest: XCTestCase {
+    
+    private var channel: EmbeddedChannel!
+    private var encoderUnderTest: LengthFieldPrepender!
+    
+    override func setUp() {
+        self.channel = EmbeddedChannel()
+    }
+
+    func testEncodeWithUInt8HeaderWithData() throws {
+        
+        self.encoderUnderTest = LengthFieldPrepender(lengthFieldLength: .one,
+                                                     lengthFieldEndianness: .little)
+
+        try? self.channel.pipeline.add(handler: self.encoderUnderTest).wait()
+        
+        let dataBytes: [UInt8] = [10, 20, 30, 40]
+        
+        var buffer = self.channel.allocator.buffer(capacity: dataBytes.count)
+        buffer.write(bytes: dataBytes)
+        
+        try self.channel.writeAndFlush(buffer).wait()
+        
+        if case .some(.byteBuffer(var headerBuffer)) = self.channel.readOutbound() {
+            
+            let outputData = headerBuffer.readBytes(length: headerBuffer.readableBytes)
+            XCTAssertEqual([UInt8(dataBytes.count)],  outputData)
+            
+            let additionalData = headerBuffer.readBytes(length: 1)
+            XCTAssertNil(additionalData)
```

Comment:
This assertion is always true by the definition of `readBytes` with a length of `readableBytes`. This will consume all of the data in the buffer, so this subsequent read must fail. We may as well remove it.